app/src/main/java/mozilla/lockbox/store/RouteStore.kt

```diff
@@ -36,14 +45,40 @@ class RouteStore(
             .map(this::dataStoreToRouteActions)
             .filterNotNull()
             .subscribe(routes)
+
+        // do we want to onboard if they disconnect their account and
+        // then re-login?
+        dataStore.state
+            .subscribe { state ->
+                when (state) {
+                    is DataStore.State.Unprepared -> onboardingState.onNext(true)
+                    else -> onboardingState.onNext(false)
+                }
+            }
+            .addTo(compositeDisposable)
```

Comment:
See how this is done above:

```kotlin
dataStore.state
    .map { it == DataStore.State.Unprepared }
    .subscribe(onboardingState)
```

Then, no need for a `compositeDisposable`.

Reply:
I'm resurrecting James' comment here; this has not been resolved yet